### PR TITLE
session: migrate test-infra to testify for tidbAsLibrarySuite

### PR DESCRIPTION
**Pull request by tisonkun**
Issue Number: ref #28328

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

**Comment by tisonkun** [comment]
@xhebox @bb7133 weird, testPessimisticSuite keep failing, while I think this patch should not touch them.

**Comment by disksing** [comment]
seems you forgot to import `pingcap/check`.

File: session/tidb_library_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	"testing"
+
+	"github.com/docker/go-units"
+	"github.com/pingcap/tidb/session"
+	"github.com/pingcap/tidb/store/mockstore"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMemoryLeak(t *testing.T) {
+	t.Skip("hack")
```
Comment (tisonkun) [inline session/tidb_library_test.go:28]: It's my fault to keep this line before clicking `merge`. Trying to re-enable it...

**Pull request merged**